tests/test_internals/test_roi_extractors_module.py

```diff
@@ -482,6 +482,289 @@ def test_add_two_plane_segmentation(self):
         assert second_plane_segmentation.description == second_plane_segmentation_description
 
 
+class TestAddFluorescenceTraces(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.num_rois = 10
+        cls.num_frames = 20
+        cls.num_rows = 25
+        cls.num_columns = 20
+
+        cls.session_start_time = datetime.now().astimezone()
+
+        cls.fluorescence_name = "fluorescence"
+        cls.dff_name = "dff"
+
+        cls.raw_roi_response_series_metadata = dict(
+            name="RoiResponseSeries",
+            description="raw fluorescence signal",
+        )
+
+        cls.dff_roi_response_series_metadata = dict(
+            name="Dff",
+            description="relative (df/f) fluorescence signal",
+        )
+
+        cls.deconvolved_roi_response_series_metadata = dict(
+            name="Deconvolved",
+            description="deconvolved fluorescence signal",
+        )
+
+        cls.neuropil_roi_response_series_metadata = dict(
+            name="Neuropil",
+            description="neuropil fluorescence signal",
+            unit="test_unit",
+        )
+
+    def setUp(self):
+        self.segmentation_extractor = generate_dummy_segmentation_extractor(
+            num_rois=self.num_rois,
+            num_frames=self.num_frames,
+            num_rows=self.num_rows,
+            num_columns=self.num_columns,
+        )
+        self.nwbfile = NWBFile(
+            session_description="session_description",
+            identifier="file_id",
+            session_start_time=self.session_start_time,
+        )
+
+        self.metadata = dict(Ophys=dict())
+
+        fluorescence_metadata = dict(
+            Fluorescence=dict(
+                name=self.fluorescence_name,
+                roi_response_series=[
+                    self.raw_roi_response_series_metadata,
+                    self.deconvolved_roi_response_series_metadata,
+                    self.neuropil_roi_response_series_metadata,
+                ],
+            )
+        )
+
+        dff_metadata = dict(
+            DfOverF=dict(
+                name=self.dff_name,
+                roi_response_series=[self.dff_roi_response_series_metadata],
+            )
+        )
+
+        self.metadata["Ophys"].update(fluorescence_metadata)
+        self.metadata["Ophys"].update(dff_metadata)
+
+        self.ophys = get_module(self.nwbfile, "ophys")
```

Comment:
Having this in the setup means that all your tests are done in an environment where the ophys module is always available, isn't not having this module available some situation that we want to test for?

Reply:
yeah, you're right, I just wanted to reduce code repetition.